### PR TITLE
Fix New-BcNuGetPackage for Runtime 3.0 (BC14)

### PR DESCRIPTION
**Pull request by sirhc101**
BC14 (runtime 3.0) does not use `id` as JSON key for the dependencies. Instead `appId` is used. `New-BcNuGetPackage` expects `id` key, which results in the following error:
```cmd
2024-01-24T09:17:47.2437517Z ##[error]ForEach-Object : Die Eigenschaft "id" wurde für dieses Objekt nicht gefunden. Vergewissern Sie sich, dass die 
Eigenschaft vorhanden ist.
In C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.4\NuGet\New-BcNuGetPackage.ps1:192 Zeichen:37
+             $appJson.dependencies | ForEach-Object {
+                                     ~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [ForEach-Object], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict,Microsoft.PowerShell.Commands.ForEachObjectCommand
```

Additionally the description of `packageId` is not correct anymore. In theory you can pass anything there, but in case of dependencies existing the template to replace placeholders is required, otherwise the ID will be the same for all dependencies.

### CODE REVIEW

File: NuGet/New-BcNuGetPackage.ps1 [hunk 1]
```diff
@@ -190,10 +190,15 @@ Function New-BcNuGetPackage {
         $XmlObjectWriter.WriteStartElement("dependencies")
         if ($appJson.PSObject.Properties.Name -eq 'dependencies') {
             $appJson.dependencies | ForEach-Object {
-                $id = CalcPackageId -packageIdTemplate $dependencyIdTemplate -publisher $_.publisher -name $_.name -id $_.id -version $_.version.replace('.','-')
+                if ($_.id) {
+                    $dependencyId = $_.id
+                } else {
+                    $dependencyId = $_.appId
+                }
+                $id = CalcPackageId -packageIdTemplate $dependencyIdTemplate -publisher $_.publisher -name $_.name -id $dependencyId -version $_.version.replace('.','-')
```
[the remark this answers is not in the document]
Comment (sirhc101) [reply NuGet/New-BcNuGetPackage.ps1:198]: Yes, you are right, that was my fault during testing.
Fixed it.

File: NuGet/New-BcNuGetPackage.ps1 [hunk 2]
```diff
@@ -190,10 +190,15 @@ Function New-BcNuGetPackage {
         $XmlObjectWriter.WriteStartElement("dependencies")
         if ($appJson.PSObject.Properties.Name -eq 'dependencies') {
             $appJson.dependencies | ForEach-Object {
-                $id = CalcPackageId -packageIdTemplate $dependencyIdTemplate -publisher $_.publisher -name $_.name -id $_.id -version $_.version.replace('.','-')
+                if ($_.id) {
+                    $dependencyId = $_.id
```
[the remark this answers is not in the document]
Comment (sirhc101) [reply NuGet/New-BcNuGetPackage.ps1:194]: No it's not.

image

[the remark this answers is not in the document]
Comment (freddydk) [reply NuGet/New-BcNuGetPackage.ps1:194]: Also with Set-StrictMode -version 2.0 and $erroractionPreference = 'stop'
Example (from another file):
`if ($_.PSObject.Properties.name -eq 'AppId') { $_.AppId } else { $_.Id }
`

[the remark this answers is not in the document]
Comment (sirhc101) [reply NuGet/New-BcNuGetPackage.ps1:194]: That's new for me! Thanks @freddydk  I was not aware of this StrictMode thing.

Fixed it
```ps1
         if ($appJson.PSObject.Properties.Name -eq 'dependencies') {
            $appJson.dependencies | ForEach-Object {
                if ($_.PSObject.Properties.name -eq 'id') {
                    $dependencyId = $_.id
                } else {
                    $dependencyId = $_.appId
                }
                $id = CalcPackageId -packageIdTemplate $dependencyIdTemplate -publisher $_.publisher -name $_.name -id $dependencyId -version $_.version.replace('.','-')
                $XmlObjectWriter.WriteStartElement("dependency")
                $XmlObjectWriter.WriteAttributeString("id", $id)
                $XmlObjectWriter.WriteAttributeString("version", $_.version)
                $XmlObjectWriter.WriteEndElement()
            }
        }
```

**Review by freddydk — APPROVED** [review]

**Pull request merged**